Add index block returned error messages (#149765)
elastic/elasticsearch: docs/reference/elasticsearch/index-settings/index-block.md

$$$index-blocks-read-only$$$

`index.blocks.read_only` {applies_to}`serverless: all`

> :   Set to `true` to make the index and index metadata read only, `false` to allow writes and metadata changes. The `cluster_block_exception` message will return `FORBIDDEN/12/index read-only / allow delete (api)`.

Reviewer: what does this mean? what about this setting would cause an error to trigger? is the idea that "when someone tries to write / edit metadata" ? we should tie action to error

Author: Apologies I won't phrase this well. It's a "you put a protective block (on purpose)" error but not a "something is wrong" error. 

So the expectation is "FYI you blocked this from occurring on purpose so some part of your setup vs current API attempt is mismatched" but not that there's inherently a problem with the config/message that you need to fix, but that it protected you by blocking the API request. 

However, users frequently google these messages because of config-vs-intent mismatch, so I wanted them to be searchable in our docs.

Reviewer: I think we still need to make the trigger of the error clear. "when x state is reached, ... "

Author: If it works, I've expanded it to say 

> While block is enabled, the `cluster_block_exception` message will return `XXXX`.

> :   Set to `true` to make the index and index metadata read only, `false` to allow writes and metadata changes. While block is enabled, the `cluster_block_exception` message will return `FORBIDDEN/12/index read-only / allow delete (api)`.

Reviewer: are you sure that this message and the one below should be the same (both with `allow delete`)?

```suggestion
:   Set to `true` to make the index and index metadata read only, `false` to allow writes and metadata changes. 

    When this block is enabled and writes or metadata changes are attempted, the `cluster_block_exception` message returns `FORBIDDEN/12/index read-only / allow delete (api)`.
```

Author: Thanks for noting, copy-pasta'd wrong. Combining wording suggestion with right message text:

```suggestion
:   Set to `true` to make the index and index metadata read only, `false` to allow writes and metadata changes. 

    When this block is enabled and writes or metadata changes are attempted, the `cluster_block_exception` message returns `FORBIDDEN/5/index read-only (api)`.
```

`index.blocks.read_only_allow_delete`

> :   Similar to `index.blocks.write`, except that you can delete the index when this block is in place. Do not set or remove this block yourself. The [disk-based shard allocator](/reference/elasticsearch/configuration-reference/cluster-level-shard-allocation-routing-settings.md#disk-based-shard-allocation) sets and removes this block automatically according to the available disk space. While block is enabled, the `cluster_block_exception` message will return `FORBIDDEN/12/index read-only / allow delete (api)`.

Reviewer: ```suggestion
:   Similar to `index.blocks.write`, except that you can delete the index when this block is in place. Do not set or remove this block yourself. The [disk-based shard allocator](/reference/elasticsearch/configuration-reference/cluster-level-shard-allocation-routing-settings.md#disk-based-shard-allocation) sets and removes this block automatically according to the available disk space. 

    When this block is enabled and writes or metadata changes are attempted, the `cluster_block_exception` message returns `FORBIDDEN/12/index read-only / allow delete (api)`.
```

$$$index-blocks-read$$$
`index.blocks.read` {applies_to}`serverless: all`

> :   Set to `true` to disable read operations against the index. While block is enabled, the `cluster_block_exception` message will return `FORBIDDEN/5/index read-only (api)`.

Reviewer: ```suggestion
:   Set to `true` to disable read operations against the index. 

    When this block is enabled and a read operation is attempted, the `cluster_block_exception` message returns `FORBIDDEN/5/index read-only (api)`.
```

Author: ```suggestion
:   Set to `true` to disable read operations against the index. 

    When this block is enabled and a read operation is attempted, the `cluster_block_exception` message returns `FORBIDDEN/7/index read (api)`.
```

$$$index-blocks-write$$$

`index.blocks.write` {applies_to}`serverless: all`

> :   Set to `true` to disable data write operations against the index. Unlike `read_only`, this setting does not affect metadata. For instance, you can adjust the settings of an index with a `write` block, but you cannot adjust the settings of an index with a `read_only` block. While block is enabled, the `cluster_block_exception` message will return `FORBIDDEN/8/index write (api)`.

Reviewer: ```suggestion
:   Set to `true` to disable data write operations against the index. Unlike `read_only`, this setting does not affect metadata. For instance, you can adjust the settings of an index with a `write` block, but you cannot adjust the settings of an index with a `read_only` block. 

    When this block is enabled and a write operation is attempted, the `cluster_block_exception` message returns `FORBIDDEN/8/index write (api)`.
```

$$$index-blocks-metadata$$$
`index.blocks.metadata` {applies_to}`serverless: all`

> :   Set to `true` to disable index metadata reads and writes. While block is enabled, the `cluster_block_exception` message will return `FORBIDDEN/9/index metadata (api)`.

Reviewer: ```suggestion
:   Set to `true` to disable index metadata reads and writes. 

   When this block is enabled and a read or write operation is attempted, the `cluster_block_exception` message returns `FORBIDDEN/9/index metadata (api)`.
```
